File: python/day01/day01.py
from typing import List
import sys

def parseInput() -> List[int]:
    result = []
    with open("day01/input.txt", 'r') as file:
        for line in file:
            result.append(int(line))
    return result

def computeIncreases(input: List[int]) -> int:
    increases = 0
    # no number in the input should be greater than this :)
    prev = sys.maxsize
    for i in input:
        if i > prev:
            increases += 1
        prev = i
    return increases

def getTriples(input: List[int]) -> List[int]:
    result = []
    index = 0
    while index < len(input):
        if index+3 > len(input):
            # return early we can't form another triple
            return result

        i1 = input[index]
        i2 = input[index+1]
        i3 = input[index+2]
        result.append(i1+i2+i3)
        index += 1
    return result

def part1(input: List[int]) -> int:
    return computeIncreases(input)

def part2(input: List[int]) -> int:
    return computeIncreases(getTriples(input))

if __name__ == '__main__':
    input = parseInput()
    print(part1(input))
    print(part2(input))

File: typescript/day01/day01.ts
import * as fs from 'fs';
import * as path from 'path';

function readInput(filename: string): number[] {
    let file = fs.readFileSync(filename, "utf-8")
    return file.split("\n").map((line) => parseInt(line))
}

function getTriples(input: number[]): number[] {
    let result: number[] = [];
    input.forEach((value, index) => {
        result.push(value+input[index+1]+input[index+2]);
    })
    return result
}

function getIncreases(input: number[]): number {
    let prev = Number.MAX_SAFE_INTEGER;
    let increases = 0;
    input.forEach((value) => {
        if (value > prev) {
            increases++
        }
        prev = value
    });
    return increases
}

function main() {
    const filename = path.join(__dirname, '..', '..', 'inputs/day01.txt');
    const input = readInput(filename);

    console.log("Part 1: " + getIncreases(input));

    const triples = getTriples(input)
    console.log("Part 2: " + getIncreases(triples))
}

main()